### DISCUSSION TITLE
Has the extraction of text-related feature been implement?

### DISCUSSION THREAD

**Discussion by Liuhaai** [vitrivr/cineast#184] [Q&A] [answered]
It seems that the extractor "OCRSearch" and "DescriptionTextSearch" don't work when I do extract operation on video files. 
Here are the errors :
> 2021-03-16 12:36:04.791 [pool-4-thread-4] FATAL o.v.c.s.r.ExtractionTask - EXTRACTION ERROR in OCRSearch: for segmentID v_0000003_2 
java.lang.UnsupportedOperationException: Not supported by default
	at org.vitrivr.cineast.core.features.abstracts.AbstractTextRetriever.processSegment(AbstractTextRetriever.java:85)
	at org.vitrivr.cineast.standalone.runtime.ExtractionTask.run(ExtractionTask.java:28)

The error output is come from this code in AbstractTextRetriever.java:
```
  public void processSegment(SegmentContainer shot) {
    throw new UnsupportedOperationException("Not supported by default");
  }
```

**Comment by lucaro** [comment, marked as answer]
No, neither of these is currently implemented, they rely on external data. In the case of the description search, this can't be implemented as an extractor, since it targets descriptions which may accompany a video file, such as what you find on youtube or vimeo, rather than being part of the video itself.
